python/lsst/analysis/drp/scatterPlot.py

```diff
@@ -0,0 +1,532 @@
+        if np.any(catPlot["sourceType"] == 1):
+            sideHist.hist(ysStars[np.isfinite(ysStars)], bins=bins, color="C0", histtype="step",
+                          orientation="horizontal", log=True)
+        sideHist.axes.get_yaxis().set_visible(False)
+        sideHist.set_xlabel("Number")
+
+        # Corner plot of patches showing summary stat in each
+        axCorner = plt.gcf().add_subplot(gs[0, -1])
+        axCorner.yaxis.tick_right()
+        axCorner.yaxis.set_label_position("right")
+        axCorner.xaxis.tick_top()
+        axCorner.xaxis.set_label_position("top")
+
+        patches = []
+        colors = []
+        for dataId in sumStats.keys():
+            (corners, stat) = sumStats[dataId]
+            ra = corners[0][0].asDegrees()
+            dec = corners[0][1].asDegrees()
+            xy = (ra, dec)
+            width = corners[2][0].asDegrees() - ra
+            height = corners[2][1].asDegrees() - dec
+            patches.append(Rectangle(xy, width, height))
+            colors.append(stat)
+            ras = [ra.asDegrees() for (ra, dec) in corners]
+            decs = [dec.asDegrees() for (ra, dec) in corners]
+            axCorner.plot(ras + [ras[0]], decs + [decs[0]], "k", lw=0.5)
+            cenX = ra + width / 2
+            cenY = dec + height / 2
+            if dataId != "tract":
+                axCorner.annotate(dataId, (cenX, cenY), color="k", fontsize=5, ha="center", va="center")
+
+        cmapUse = plt.cm.coolwarm
+        # Set the bad color to transparent and make a masked array
+        cmapUse.set_bad(color="none")
+        colors = np.ma.array(colors, mask=np.isnan(colors))
+        collection = PatchCollection(patches, cmap=cmapUse)
+        collection.set_array(colors)
+        axCorner.add_collection(collection)
+
+        axCorner.set_xlabel("R.A. (deg)", fontsize=8)
+        axCorner.set_ylabel("Dec. (deg)", fontsize=8)
+        axCorner.tick_params(axis="both", labelsize=8)
+
+        # Add a colorbar
+        divider = make_axes_locatable(axCorner)
+        cax = divider.append_axes("left", size="14%")
+        plt.colorbar(collection, cax=cax, orientation="vertical")
+        cax.yaxis.set_ticks_position("left")
+        for label in cax.yaxis.get_ticklabels():
+            label.set_bbox(dict(facecolor="w", ec="none", alpha=0.5))
+            label.set_fontsize(8)
+        cax.text(0.6, 0.5, "Median Value", color="k", rotation="vertical", transform=cax.transAxes,
+                 horizontalalignment="center", verticalalignment="center", fontsize=8)
+
+        plt.draw()
+
+        # Add a legend
+        for (i, newLabel) in enumerate(highStats):
+            linesForLegend[i].set_label(newLabel)
+        highStatThresh = self.config.highSnStatisticSelectorActions.statSelector.threshold
+        lowStatThresh = self.config.lowSnStatisticSelectorActions.statSelector.threshold
+        highLegendTitle = "S/N > {} Stats \n({} < {:0.2f})".format(highStatThresh, xCol, approxHighMag)
```

Comment:
Wait- but this line now conflicts with my understanding of the difference between `xCol` and `magCol`.  `approxHighMag` was computed from the `magCol` column, but now the legend is reporting as `xCol` less than something.  I think either `xCol` here should be `magCol`, or `magCol` itself should just become `xCol` (as I said above, perhaps an example illustrating an example of the need for `magCol` in addition to `xCol` would be instructive).

Reply:
Good catch, that should be magCol.